### ISSUE TITLE
Iterating `bytearray` yields `System.Byte` objects

### ISSUE THREAD

**Issue by pekkaklarck** [IronLanguages/main#1237]
If items in a bytearray are accessed with an index you get `int` objects as expected. Iterating a bytearray yields `System.Byte` objects.

```python
IronPython 2.7.5 (2.7.5.0) on .NET 4.0.30319.34209 (32-bit)
Type "help", "copyright", "credits" or "license" for more information.
>>> ba = bytearray(b'abc')
>>> ba[0]
97
>>> list(ba)[0]
<System.Byte object at 0x000000000000002B [97]>
>>>
```

**Comment by simplicbe** [comment]
The reason for this is, that bytearray inherit from `IList<byte>`. So if you iterate over it, it returns an byte-array. This indexing is implemented like this:

```
public object this[int index] {
    get {
        lock (this) {
            return ScriptingRuntimeHelpers.Int32ToObject((int)_bytes[PythonOps.FixIndex(index, _bytes.Count)]);
           }
       }
       
    /// ....
```

And Int32ToObject makes the following: 
> Gets a singleton boxed value for the given integer if possible, otherwise boxes the integer.

So if you need to get an integer at this point, you will have to cast to to an integer or make some changes in **ByteArray.cs**. I'm not sure what the default behaviour should be.

Hope this helps.

**Comment by jdhardy** [comment]
In all cases, the default behaviour should match what CPython does. We can
add an extra method to iterate over bytes but the default should be ints if
that's what CPython does.

**Comment by simplicbe** [comment]
Thank you. Ok, i create a pull request with the changes.

**Comment by slide** [comment]
Fixed in https://github.com/IronLanguages/main/pull/1240 ?

**Comment by simplicbe** [comment]
Yes, #1240 should fix this issue.

**Comment by slide** [comment]
Fixed in #1240
